inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,318 @@
+
+    def extractor(field):
+        def decorator(decorated_function):
+            store[field] = decorated_function
+            return decorated_function
+        return decorator
+    extractor.store = store
+    return extractor
+
+
+def bibtex_document_type(doc_type, obj):
+    """
+    Args:
+        doc_type (text_type): INSPIRE document type.
+        obj (dict): literature record.
+
+    Returns:
+        text_type: bibtex document type for the given INSPIRE entry.
+    """
+    DOCUMENT_TYPE_MAP = {
+        'article': 'article',
+        'book': 'book',
+        'book chapter': 'inbook',
+        'conference paper': 'inproceedings',
+        'proceedings': 'proceedings',
+        'report': 'techreport',
+        'note': 'article',
+        # theses handled separately due to masters/phd distinction
+    }
+    if doc_type in DOCUMENT_TYPE_MAP:
+        return DOCUMENT_TYPE_MAP[doc_type]
+    # Theses need special treatment, because bibtex differentiates between their types:
+    elif doc_type == 'thesis' and obj['thesis_info']['degree_type'] in ('phd', 'habilitation'):
+        return 'phdthesis'
+    # Other types of theses (other, bachelor, laurea) don't have separate types in bibtex:
+    # We will use the type field (see `get_type`) to indicate the type of diploma.
+    elif doc_type == 'thesis':
+        return 'mastersthesis'
+    return 'misc'
+
+
+def bibtex_type_and_fields(data):
+    """
+    Args:
+        inspire record
+
+    Returns:
+         tuple: bibtex document type and fields
+    """
+    # TODO: Establish a better method with which we choose the bibtex type if there is more that one inspire doc type
+    bibtex_doc_types = [bibtex_document_type(doc_type, data) for doc_type in data['document_type']] + ['misc']
+    # Preference towards article, as it's more prestigious to have sth published:
+    chosen_type = 'article' if 'article' in bibtex_doc_types else bibtex_doc_types[0]
+    return chosen_type, FIELDS_FOR_ENTRY_TYPE[chosen_type] + COMMON_FIELDS_FOR_ENTRIES
+
+
+def get_authors_with_role(authors, role):
+    """
+    Extract names of people from an authors field given their roles.
+    Args:
+        authors: authors field of the record.
+        role: string specifying the role 'author', 'editor', etc.
+
+    Returns:
+        list of names of people
+    """
+    return [author['full_name'] for author in authors if role in author.get('inspire_roles', ['author'])]
+
+
+def get_country_name_by_code(code, default=None):
+    """
+    Args:
+        code (str): country code in INSPIRE 2 letter format based on ISO 3166-1 alpha-2
+        default: value to be returned if no country of a given code exists
+
+    Returns:
+        text_type: name of a country, or a code if no country found
+    """
+    try:
+        return Locale('en').territories[code]
+    except KeyError:
+        return default
+
+
+def get_date(data, doc_type):
+    publication_year = get_value(data, 'publication_info[0].year')
```

Comment:
As we discussed on the board, in general `foo[0].bar` should be `foo.bar[0]`.

Reply:
I was wary of a situation like this happening:

```python
publication_info = [
    {'journal_title': 'A journal'},
    {'journal_title': 'Another journal', 'journal_issue': '12'},
    {'journal_title': 'Yet another one', 'year': 2016},
]
```

In which case the final record will have mismatched data. Granted, what I have now is a bad solution, I could try to pick the "fullest" publication info entry?

Reply:
> I could try to pick the "fullest" publication info entry?

This sounds much better :+1:

Reply:
you should also exclude `material` that is not `publication` or empty, as you don't want to take errata here. By the way, are you handling errata like legacy https://inspirehep.net/record/374836/export/hx ?

Reply:
Working on it. Only errata should be handled like this, or should addenda too?

Reply:
:+1: addenda too